### PR TITLE
added wandb logger and extended when to log

### LINKED ISSUE #362: [REQUEST] Support WandB logging
**Is your feature request related to a problem? Please describe.**
I would like to use WandB to log my runs

**Describe the solution you'd like**
For this I need a WandBLoggerAdapter that uses the wandb python api to write metrics.
I'm curious if this is something where you also see added value. 
I would love to help; I already made a version I'm using myself locally.

### PR DESCRIPTION
**Pull request by claudius-kienle**
I added a wandb logger adapter and also added to new parameters to customize when to log.

This will close #362

### CODE REVIEW

File: d3rlpy/logging/wandb_adapter.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
+import wandb
+from typing import Any, Dict, Optional
+from .logger import LoggerAdapter, LoggerAdapterFactory, SaveProtocol
+
```
Comment (takuseno) [inline d3rlpy/logging/wandb_adapter.py:4]: Please add `__all__` in this line.
```py
__all__ = ["LoggerWanDBAdapter", "WanDBAdapterFactory"]
```
And, then, please add an import to this file:
https://github.com/takuseno/d3rlpy/blob/master/d3rlpy/logging/__init__.py

File: d3rlpy/logging/wandb_adapter.py [hunk 2]
```diff
@@ -0,0 +1,73 @@
+import wandb
```
Comment (takuseno) [inline d3rlpy/logging/wandb_adapter.py:1]: To allow users to use d3rlpy without installing wandb, can you import the package just like here?
https://github.com/takuseno/d3rlpy/blob/06dd7a7698bf68af0645afeb62f148aad921a9bd/d3rlpy/logging/tensorboard_adapter.py#L30

File: d3rlpy/logging/wandb_adapter.py [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    r"""WandB Logger Adapter class.
+
+    This class logs data to Weights & Biases (WandB) for experiment tracking.
+
+    Args:
+        experiment_name (str): Name of the experiment.
+    """
+
+    def __init__(self, project: Optional[str] = None, experiment_name: Optional[str] = None):
+        self.run = wandb.init(project=project, name=experiment_name)
```
Comment (takuseno) [inline d3rlpy/logging/wandb_adapter.py:16]: This is just a question. I'm not familiar with wandb, but if `project` is `None`, what would happen?

File: d3rlpy/algos/qlearning/base.py [hunk 4]
```diff
@@ -360,6 +360,8 @@ def fit(
         n_steps_per_epoch: int = 10000,
         experiment_name: Optional[str] = None,
         with_timestamp: bool = True,
+        logging_steps: int = 500,
+        logging_strategy: str = "epoch",
```
Comment (takuseno) [inline d3rlpy/algos/qlearning/base.py:364]: This is a good idea! I have one request to change how we switch `epoch` and `steps`. I'd like to use Enum to specify this option. In this way, could you add `LoggingStrategy` enum class to `constants.py` as follows?
https://github.com/takuseno/d3rlpy/blob/06dd7a7698bf68af0645afeb62f148aad921a9bd/d3rlpy/constants.py#L47

Then, it's easier for users to see what kind of options are available.

**Review by takuseno — CHANGES_REQUESTED** [review]
@claudius-kienle Thank you for your contribution! This is awesome! I have some minor comments that I'd like you to address before we merge this. Please take a look. Thanks again!

File: d3rlpy/logging/wandb_adapter.py (hunk 3 above)
Comment (claudius-kienle) [reply d3rlpy/logging/wandb_adapter.py:16]: If project is `None`, wandb will save this run in a new project, whose name is inferred from the relative directory the script resides at. E.g. if you start a training script from `scripts/training/train_sac.py` the WanDB project will be called `REPO_NAME-scripts_training_train_sac`, where REPO_NAME is the folder name of the git repository

**Comment by claudius-kienle** [comment]
@takuseno Thank you for the review. All your comments should now be included.

File: d3rlpy/constants.py [hunk 5]
```diff
@@ -47,3 +47,8 @@ class ActionSpace(Enum):
 class PositionEncodingType(Enum):
     SIMPLE = "simple"
     GLOBAL = "global"
+
+
+class LoggingStrategyEnum(Enum):
```
Comment (takuseno) [inline d3rlpy/constants.py:52]: Could you rename this to `LoggingStrategy`?

**Review by takuseno — CHANGES_REQUESTED** [review]
Thanks for the change! It looks good to me. I have one minor request that I want you to address!

File: d3rlpy/constants.py (hunk 5 above)
Comment (claudius-kienle) [reply d3rlpy/constants.py]: Of course. I just changed it.

**Review by takuseno — APPROVED** [review]
Thanks! LGTM. I see formattting errors, but I'll follow up.

**Pull request merged**